ExadelTimeTrackingSystem/ExadelTimeTrackingSystem.Data/Repositories/TaskRepository.cs

```diff
@@ -48,5 +59,35 @@ public Task BulkCreateAsync(List<Models.Task> tasks, CancellationToken cancellat
             cancellationToken.ThrowIfCancellationRequested();
             return GetCollection<Models.Task>().InsertManyAsync(tasks);
         }
+
+        public async Task<Dictionary<DayOfWeek, int>> GetHoursByDatesAsync(DateTime date, Guid employeeId, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            List<DateTime> dateList = new List<DateTime>();
+            List<string> testList = new List<string>();
+            var startDate = date;
+            var endDate = date.DeepCopy().AddDays(6);
+            for (int i = 0; i < 7; i++)
+            {
+                dateList.Add(date);
+                date = date.AddDays(1);
+            }
+
+            var filterBuilder = Builders<Models.Task>.Filter;
+            var employeeFilter = filterBuilder.Eq(t => t.EmployeeId, employeeId);
+            var dateFilter = filterBuilder.In(t => t.Date, dateList);
+            var testFilter = filterBuilder.Where(t => t.Date >= startDate && t.Date <= endDate);
+
+            var hours = await GetCollection<Models.Task>().Aggregate().Match(testFilter & employeeFilter).Group(
+                t => t.Date,
+                group => new
+                {
+                    date = group.Key,
+                    Total = group.Sum(t => t.HoursSpent),
+                }).ToListAsync();
+
+            var hourDictionary = hours.ToDictionary(t => t.date.DayOfWeek, t => t.Total);
```

Comment:
better to return as a key of dictionary just date, and call DayOfWeek in service, because response from repository should be universal, in case if other services can use it for example (e.g. we have another service which returns not Monday: 10, but 20/06/2022: 10). and DayOfWeek again is more about business logic, but repository shouldn't be aware of format in which you want to return these hours for user